test/integration/component/test_simultaneous_volume_attach.py

```diff
@@ -0,0 +1,255 @@
+            cmd.jobid = jobid
+            timeout = asyncTimeout
+            async_response = FAILED
+            while timeout > 0:
+                async_response = apiclient.queryAsyncJobResult(cmd)
+                if async_response != FAILED:
+                    job_status = async_response.jobstatus
+                    if job_status in [JOB_CANCELLED,
+                                  JOB_SUCCEEDED]:
+                        break
+                    elif job_status == JOB_FAILED:
+                        raise Exception("Job failed: %s" \
+                                    % async_response)
+                time.sleep(5)
+                timeout -= 5
+                self.debug("=== JobId: %s is Still Processing, "
+                              "Will TimeOut in: %s ====" % (str(jobid),
+                                                           str(timeout)))
+            return async_response
+        except Exception as e:
+            self.debug("==== Exception Occurred for Job: %s ====" %
+                  str(e))
+            return FAILED
+
+    @attr(tags = ["advanced", "advancedns", "basic"], required_hardware="true")
+    def test_attach_multiple_volumes(self):
+        """Attach multiple Volumes simultaneously to a Running VM
+        """
+        # Validate the following
+        # 1. All data disks attached successfully without any exception
+
+        self.debug(
+                "Attaching volume (ID: %s) to VM (ID: %s)" % (
+                                                    self.volume1.id,
+                                                    self.virtual_machine.id
+                                                    ))
+        vol1_jobId = self.attach_volume(self.apiClient, self.virtual_machine.id,self.volume1)
+
+        self.debug(
+                "Attaching volume (ID: %s) to VM (ID: %s)" % (
+                                                    self.volume2.id,
+                                                    self.virtual_machine.id
+                                                    ))
+        vol2_jobId = self.attach_volume(self.apiClient,self.virtual_machine.id, self.volume2)
+
+        self.debug(
+                "Attaching volume (ID: %s) to VM (ID: %s)" % (
+                                                    self.volume3.id,
+                                                    self.virtual_machine.id
+                                                    ))
+        vol3_jobId = self.attach_volume(self.apiClient,self.virtual_machine.id, self.volume3)
+
+        self.debug(
+                "Attaching volume (ID: %s) to VM (ID: %s)" % (
+                                                    self.volume4.id,
+                                                    self.virtual_machine.id
+                                                    ))
+        vol4_jobId = self.attach_volume(self.apiClient,self.virtual_machine.id, self.volume4)
+
+        self.query_async_job(self.apiClient,vol1_jobId.jobid)
```

Comment:
We are not catching the return value of the query_asyn_job method. We can fail the test here if the return value is FAILED. Feel free to comment if you don't agree with this.